decidim-elections/app/services/decidim/votings/census_vote_flow.rb

```diff
@@ -38,15 +38,33 @@ def login_path(vote_path)
         EngineRouter.main_proxy(election.component.participatory_space).voting_login_path(election_id: election.id, vote_path: vote_path)
       end
 
+      def questions_for(election)
+        if ballot_style.present?
+          ballot_style.questions.where(election: election)
+        else
+          election.questions
+        end
+      end
+
+      def ballot_style_id
+        ballot_style&.slug
+      end
+
       private
 
+      def ballot_style
+        return @ballot_style if defined?(@ballot_style)
+
+        @ballot_style = datum&.ballot_style
+      end
```

Comment:
💭 Maybe this be rewritten in a more readable form:

```suggestion
      def ballot_style
        @ballot_style ||= datum&.ballot_style
      end
```